### DISCUSSION TITLE
FYTA Timeout Problems

### DISCUSSION THREAD

**Discussion by edgegit** [dontinelli/fyta-custom_component#55] [Q&A]
Just recently started to have lot's of issues with the FYTA integration and sensors becoming periodically unavailable. Log's showing this. Shall I open up a bug ticket? 

Logger: homeassistant.components.fyta.coordinator
Source: helpers/update_coordinator.py:386
integration: FYTA (documentation, issues)
First occurred: 24 August 2024 at 06:24:53 (595 occurrences)
Last logged: 10:37:02

Error fetching FYTA Coordinator data: Timeout occurred while connecting to Fyta-server

HA: 2024.8.2

**Comment by dontinelli** [comment]
Hi Rob
Thank you for your report. I encountered the same, but have not yet had to investigate it further. I assume it is an issue with fyta_cli and not the custom_component. It could be related to changes on the server side (introduction of remote live mode).
Can you please confirm if your sensors are still available or if all went unavailable?

**Reply by edgegit** [reply]
Hey, 

Thanks for the reply. I have 6 sensors and 3 hubs. All sensor param's become periodically unavailable. 

<img width="595" alt="Screenshot 2024-08-26 at 11 13 28" src="https://github.com/user-attachments/assets/8c21b7f5-788d-4cd5-bda8-d11b16bb5e0c">

p.s. hi from Bern ;)

**Reply by dontinelli** [reply]
You are not alone: https://github.com/home-assistant/core/issues/124642

**Reply by dontinelli** [reply]
I will try to increase timeout and hope this resolves the issue.

**Reply by someshdev** [reply]
Please increase the update interval of homeassistant.components.fyta.coordinator to a minimum of 3x the value

**Reply by dontinelli** [reply]
What is the rational behind your proposal?

**Reply by someshdev** [reply]
Our backend queries for each plant and plant list every minute. This creates a throttle. Increasing the interval is an interim solution. In the meantime we are working on an new end point for users that require specific objects. Will keep you updated

**Comment by dontinelli** [comment]
Hi @edgegit
I increased the timeout to 60 seconds in the latest release. With this some of the timeouts can be mitigated. However, the client now sometimes runs into a gateway timeout. It seems the problem is indeed with the server having longer to respond as expected...

**Reply by edgegit** [reply]
Hey @dontinelli 

Thanks for the update (both message and code). What does that mean now for the long-term fix?

**Reply by dontinelli** [reply]
I will have to further investigate, if there is anything I can do from my side. But I assume the problem is beyond my power/responsibility.

**Reply by edgegit** [reply]
Pity. The integration is not long reached platinum status :( - Thanks for the good work anyway.

**Reply by dontinelli** [reply]
Yes, indeed. FYTA is investigating the problem (https://discord.com/channels/1052218149489094676/1052218150193725452/1278285210949124159)

**Reply by dontinelli** [reply]
I noticed that the timeout is predominantly triggered by one API request. I could suppress this request. Please test latest version of CC and let me know if fix works for you.

**Reply by edgegit** [reply]
CC? - Happy to check, but where/how do we update to see these changes?

**Reply by edgegit** [reply]
Ah. Custom-Component. I see. I've not installed via HACS, but official installation.

**Reply by dontinelli** [reply]
Then you have to wait a little bit more. I prepared a PR, that hopefully will make it into 2024.09 (released on Wednesday).

**Reply by dontinelli** [reply]
I assumed you have installed the CC, as you reported it in this repository...

**Reply by dontinelli** [reply]
Should be part of 2024.9: https://github.com/home-assistant/core/pull/125065#event-14100475967

**Reply by edgegit** [reply]
Yea, sorry. I was bit confused where exactly this should be reported. Excellent about the 2024.9 release.
